### PR TITLE
[BC] Allow only `bool` tensors as mask in `masked_select`

### PR DESCRIPTION
**Pull request by malfet**
`byte` support was marked as deprecated in 1.8, so it's fine to remove this in 2.1 (or even 2.0)
Deprecation warning was added by https://github.com/pytorch/pytorch/pull/22261

Also, fix bunch of syntactic errors in comments

cc @albanD

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: aten/src/ATen/native/TensorAdvancedIndexing.cpp [hunk 1]
```diff
@@ -1899,12 +1894,12 @@ static Tensor & masked_select_out_impl_cpu(Tensor & result, const Tensor & self,
   auto mask_long_data = mask_long.data_ptr<int64_t>();
   auto mask_prefix_sum_data = mask_prefix_sum.data_ptr<int64_t>();
   // TODO: Here can only use std::partial_sum for C++14,
-  // use std::exclusive_scan when PyTorch upgrades to C++17, which have better peformance.
+  // use std::exclusive_scan when PyTorch upgrades to C++17, which have better performance.
   // std::exclusive_scan(mask_long_data, mask_long_data + mask_long.numel(), mask_prefix_sum_data, 0);
```
Comment (Skylion007) [inline aten/src/ATen/native/TensorAdvancedIndexing.cpp:1898]: @malfet can you swap out the comments to std::exclusive_scan now that we are in C++17?

Comment (malfet) [reply aten/src/ATen/native/TensorAdvancedIndexing.cpp:1898]: I'll do it (and few other changes) in followup PR (don't want to muddy this one with unrelated functional changes)

Comment (malfet) [reply aten/src/ATen/native/TensorAdvancedIndexing.cpp:1898]: https://github.com/pytorch/pytorch/pull/96217

**Pull request closed without merging**